.github/workflows/CI.yaml

```
@@ -0,0 +1,46 @@
+name: Run Tests
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master
+  push:
+    branches:
+      - "master"
+
+  # trigger on request
+  workflow_dispatch:
+
+jobs:
+  run_tests:
+    name: Run tests on ubuntu-latest with Python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8","3.9","3.10","3.11"]
+    steps:
+    - uses: actions/checkout@v3
```

```suggestion
    - uses: actions/checkout@v4
```
Checkout is on v4 now. If you don't mind a few extra dependabot PRs, use `v4.0.0` to ensure repeatability.